### PR TITLE
Improve input invalidation, custom input type declaration and transition animation performance

### PR DESCRIPTION
**Pull request by ThaUnknown**
These are all non-breaking changes, and minor improvements.

### Input invalidation:
These changes will apply the same effect as `.is-invalid` to any input that is invalid via `:invalid` pseudoselector, this happens when for example an input number is out of range, text doesnt follow regex, file has wrong extension etc, without any JS needed for validation like for example:
```html
<input type="text" pattern="[A-Za-z]{3}">
<input type="number" min="0" max="100" step="5">
```
those will invalidate when the text doesn't match its regex pattern and when the number is `<0` or `>100` or `%5!==0`, these are just a few examples tho

### Better declaration for custom input types:
TLDR, old:
```html
<div class="custom-radio">
  <input type="radio" name="radio-set-1" id="radio-1" value="radio-1">
  <label for="radio-1">Radio 1</label>
</div>
<div class="custom-radio">
  <input type="radio" name="radio-set-1" id="radio-2" checked="checked" value="radio-2">
  <label for="radio-2">Radio 2</label>
</div>
```
new:
```html
<div class="custom-radio">
  <input type="radio" name="radio-set-1" id="radio-1" value="radio-1">
  <label for="radio-1">Radio 1</label>
  <input type="radio" name="radio-set-1" id="radio-2" checked="checked" value="radio-2">
  <label for="radio-2">Radio 2</label>
</div>
```
I applied this for checkmarks, radios and switches, but the old way is still supported since these changes **_SHOULD BE_** non-breaking, but that means there's an additional useless `position: relative` inside `.custom-${type}`

### Animation performance

applied the `will-change` property to address performance issues when animating the sidebar from https://github.com/halfmoonui/halfmoon/issues/43#issuecomment-706368351
I'm kinda torn on this one because [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/will-change) says never to use this, on the other hand [Google](https://developers.google.com/web/fundamentals/design-and-ux/animations/animations-and-performance#using_the_will-change_property) says to use it on any animation, so I decided only to apply this for the very heavy sidebar animation that changes pretty much the entire site layout, but from my own testing this helps A LOT

### CODE REVIEW

**Comment by halfmoonui** [comment]
Hey, thanks a lot for the PR. I haven't gotten the chance to properly dive in, but everything looks to be good. I too am conflicted about the `will-change` property, not sure what to make of it to be honest. I am planning a new commit to the `dev` branch, so I will probably merge this before committing my own changes.

**Comment by ThaUnknown** [comment]
seems like i missed some elements like select and possibly others with ::invalid

**Comment by ksenginew** [comment]
Hi @ThaUnknown  It looks like halfmoon is not in active development. I had created an active clone named **Fullmoon**. Please refer to https://github.com/halfmoonui/halfmoon/issues/116#issuecomment-1013674632 for more info.

You can report any issues [here](https://github.com/ksenginew/fullmoon/issues). And [contribute Fullmoon](https://github.com/ksenginew/fullmoon) to add this feature.
